pykafka/pykafka/simpleconsumer.py

```diff
@@ -238,40 +247,49 @@ def fetch_offsets(self):
 
         reqs = [p.build_offset_fetch_request() for p in self._partitions.keys()]
         res = self._offset_manager.fetch_consumer_group_offsets(self._consumer_group, reqs)
-        parts_by_error = self._filter_partition_responses(res)
         error_handlers = {0: lambda op, pres: op.set_offset(pres.offset)}
-        self._handle_partition_errors(parts_by_error, error_handlers)
-        self._reset_offsets(
-            [a[0] for a in
-             parts_by_error[OffsetOutOfRangeError.ERROR_CODE]])
+        self._handle_partition_errors(res, handlers=error_handlers)
 
     def _filter_partition_responses(self, res):
         """Group partition responses by error code
 
-        This function accepts as input a FetchResponse or an OffsetFetchResponse.
+        This function accepts as input a Response instance
 
         :param res: a Response containing zero or more partition responses
-        :type res: FetchResponse or OffsetFetchResponse
+        :type res: pykafka.protocol.Response
         """
         partitions_by_error = defaultdict(list)
-        for partition_id, pres in res.topics[self._topic.name].iteritems():
-            owned_partition = self._partitions_by_id[partition_id]
-            partitions_by_error[pres.error].append((owned_partition, pres))
+        for topic_name in res.topics.keys():
+            for partition_id, pres in res.topics[topic_name].iteritems():
+                owned_partition = self._partitions_by_id[partition_id]
+                partitions_by_error[pres.error].append((owned_partition, pres))
         return partitions_by_error
 
-    def _handle_partition_errors(self, parts_by_error, handlers):
+    def _handle_partition_errors(self, res, handlers=None):
         """Call the appropriate handler for each errored partition
 
-        :param parts_by_error: partition responses grouped by error code
-            (output of self._filter_partition_responses)
-        :type parts_by_error: dict {int: iterable}
+        :param res: a Response object containing partition responses
+        :type res: pykafka.protocol.Response
         :param handlers: mapping of error code to handler
         :type handlers: dict {int: callable(owned_partition, partition_response)}
         """
+        if handlers is None:
+            handlers = {}
+        error_handlers = dict(
+            self._default_error_handlers.items() + handlers.items())
+
+        parts_by_error = self._filter_partition_responses(res)
+
         for errcode, parts in parts_by_error.iteritems():
             for owned_partition, pres in parts:
-                if errcode in handlers:
-                    handlers[errcode](owned_partition, pres)
+                if errcode in error_handlers:
+                    error_handlers[errcode](owned_partition, pres)
+
+        self._reset_offsets(
+            [a[0] for a in
+             parts_by_error[OffsetOutOfRangeError.ERROR_CODE]])
```

Comment:
This can't be  part of that handler dict because the handler function would involve appending these partitions to some shared sequence somewhere before calling `_reset_offsets`. More simply, `_reset_offsets` requires a list, which is at odds with the handler scheme. I suppose this could be moved into another function and the rest of `_handle_partition_errors` could be put in utils.